### PR TITLE
circular-buffer: add exercise

### PR DESCRIPTION
**Pull request by wolf99**
For #390

### CODE REVIEW

File: exercises/circular-buffer/src/example.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#ifndef CIRCULAR_BUFFER_H
+#define CIRCULAR_BUFFER_H
+
+#include <stdlib.h>
```
Comment (ryanplusplus) [inline exercises/circular-buffer/src/example.h]: Is this to pull in `size_t`? You can pull in `stddef.h` to get it directly (I believe that `stdlib.h` is defined to pull in `stddef.h`, but I'm not 100% certain this is portable).

Comment (wolf99) [reply exercises/circular-buffer/src/example.h]: Yep. Have updated :)

File: exercises/circular-buffer/test/test_circular_buffer.c [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+                          1, 2}, false, buffer);
+   write_values_to_buffer(1, (buffer_value_t[]) {
+                          3}, true, buffer);
+   read_values_from_buffer(2, (buffer_value_t[]) {
+                           2, 3}, buffer);
+
+   delete_buffer(buffer);
+}
+
+static void test_overwrite_replaces_oldest_item_reamining_following_read(void)
```
Comment (ryanplusplus) [inline exercises/circular-buffer/test/test_circular_buffer.c]:
```suggestion
static void test_overwrite_replaces_oldest_item_remaining_following_read(void)
```

File: exercises/circular-buffer/test/test_circular_buffer.c [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+   RUN_TEST(test_items_are_read_in_order_written);
+   RUN_TEST(test_full_buffer_cannot_be_written_to);
+   RUN_TEST(test_read_frees_capacity_for_another_write);
+   RUN_TEST(test_read_position_maintained_across_multiple_writes);
+   RUN_TEST(test_cleared_buffer_cannot_be_read);
+   RUN_TEST(test_clear_frees_capacity_for_another_write);
+   RUN_TEST(test_clear_does_nothing_on_an_empty_buffer);
+   RUN_TEST(test_overwrite_acts_like_write_on_non_full_buffer);
+   RUN_TEST(test_overwrite_replaces_oldest_item_on_full_buffer);
+   RUN_TEST(test_overwrite_replaces_oldest_item_reamining_following_read);
```
Comment (ryanplusplus) [inline exercises/circular-buffer/test/test_circular_buffer.c]:
```suggestion
   RUN_TEST(test_overwrite_replaces_oldest_item_remaining_following_read);
```

**Comment by ryanplusplus** [comment]
Looks good to me pending typo fix

File: exercises/circular-buffer/test/test_circular_buffer.c (hunk 2 above)
Comment (wolf99) [reply exercises/circular-buffer/test/test_circular_buffer.c]: D'Oh! IDE is supposed to warn me about typos like this

**Comment by wolf99** [comment]
Thanks! Happy holidays 🎄 🎅

**Pull request merged**